src/hooks/session.hook.ts

```diff
@@ -0,0 +1,27 @@
+import { useApiSession } from '../api/hooks/api-session.hook';
+import { useWalletContext } from '../contexts/wallet.context';
+
+export interface SessionInterface {
+  isMetaMaskInstalled: boolean;
+  address?: string;
+  isLoggedIn: boolean;
+  login: () => Promise<void>;
+}
+
+export function useSession(): SessionInterface {
+  const { isLoggedIn, getSignMessage, createSession } = useApiSession();
+  const { isInstalled, isConnected, address, connect, signMessage } = useWalletContext();
+
+  async function login(): Promise<void> {
+    let connectedAddress = address;
+    if (!isConnected) {
+      connectedAddress = await connect();
+    }
+    if (!connectedAddress || isLoggedIn) return;
```

Comment:
Should we throw an error, if `!connectedAddress`?

Reply:
I am not quite sure, if this would even be possible.
If a user declines connecting his MM to our exchange, it will throw an error on an another level und is handled there. But for sake of robustness we should add one here. yes. will add a TODO comment.

And then come up with a error handling concept later on